src/lib-smtp/fuzz-smtp-server.c

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2020 Catena cyber */
+
+#include "lib.h"
+#include "str.h"
+#include "array.h"
+#include "hostpid.h"
+#include "ioloop.h"
+#include "istream.h"
+#include "ostream.h"
+#include "time-util.h"
+#include "sleep.h"
+#include "connection.h"
+#include "test-common.h"
+#include "smtp-address.h"
+#include "smtp-reply-parser.h"
+#include "smtp-server.h"
+
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+static struct ioloop *ioloop_fuzz;
+static int initialized = 0;
+int fdo[2];
+int fdi[2];
+
+struct server_connection {
+    void *context;
+};
+
+
+static void server_connection_destroy(void *context)
+{
+    struct server_connection *sconn = (struct server_connection *)context;
+    i_free(sconn);
+}
+
+static void
+test_server_defaults(struct smtp_server_settings *smtp_set)
+{
+    /* server settings */
+    i_zero(smtp_set);
+    smtp_set->max_client_idle_time_msecs = 500;
+    smtp_set->max_pipelined_commands = 16;
+    smtp_set->auth_optional = TRUE;
+    smtp_set->debug = 0;
```

Comment:
debug = FALSE .. or just don't set it, since it's the default anyway

Reply:
Indeed, this is a leftover